### ISSUE TITLE
ratelimit traceback with `secure_admin_login()`

### ISSUE THREAD

**Issue by rcj4747** [pennersr/django-allauth#3831]
We upgraded to 0.63.1 and added `secure_admin_login()` per the [doc](https://docs.allauth.org/en/latest/common/admin.html) but `runserver` gives us a traceback.  I've recreated with the regular-django example in this project's source.  Here's the patch I applied (following the docs) to recreate the issue:

```$ git diff
diff --git a/examples/regular-django/example/urls.py b/examples/regular-django/example/urls.py
index ffdf8ec2..c099529d 100644
--- a/examples/regular-django/example/urls.py
+++ b/examples/regular-django/example/urls.py
@@ -2,8 +2,11 @@ from django.contrib import admin
 from django.urls import include, path
 from django.views.generic.base import TemplateView
 
+from allauth.account.views import login as secure_admin_login
+
 
 admin.autodiscover()
+admin.site.login = secure_admin_login(admin.site.login)
 
 urlpatterns = [
     path("", TemplateView.as_view(template_name="index.html")),
````

Once you create the virtualenv per `examples/regular-django/example/Readme.org` you'll get the following traceback with `runserver`:

```
$ python manage.py runserver 8999
Watching for file changes with StatReloader
Performing system checks...

Exception in thread django-main-thread:
Traceback (most recent call last):
  File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/home/me/devel/source/django-allauth/examples/regular-django/venv/lib/python3.10/site-packages/django/utils/autoreload.py", line 64, in wrapper
    fn(*args, **kwargs)
  File "/home/me/devel/source/django-allauth/examples/regular-django/venv/lib/python3.10/site-packages/django/core/management/commands/runserver.py", line 133, in inner_run
    self.check(display_num_errors=True)
  File "/home/me/devel/source/django-allauth/examples/regular-django/venv/lib/python3.10/site-packages/django/core/management/base.py", line 486, in check
    all_issues = checks.run_checks(
  File "/home/me/devel/source/django-allauth/examples/regular-django/venv/lib/python3.10/site-packages/django/core/checks/registry.py", line 88, in run_checks
    new_errors = check(app_configs=app_configs, databases=databases)
  File "/home/me/devel/source/django-allauth/examples/regular-django/venv/lib/python3.10/site-packages/django/core/checks/urls.py", line 14, in check_url_config
    return check_resolver(resolver)
  File "/home/me/devel/source/django-allauth/examples/regular-django/venv/lib/python3.10/site-packages/django/core/checks/urls.py", line 24, in check_resolver
    return check_method()
  File "/home/me/devel/source/django-allauth/examples/regular-django/venv/lib/python3.10/site-packages/django/urls/resolvers.py", line 519, in check
    for pattern in self.url_patterns:
  File "/home/me/devel/source/django-allauth/examples/regular-django/venv/lib/python3.10/site-packages/django/utils/functional.py", line 47, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/home/me/devel/source/django-allauth/examples/regular-django/venv/lib/python3.10/site-packages/django/urls/resolvers.py", line 738, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
  File "/home/me/devel/source/django-allauth/examples/regular-django/venv/lib/python3.10/site-packages/django/utils/functional.py", line 47, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/home/me/devel/source/django-allauth/examples/regular-django/venv/lib/python3.10/site-packages/django/urls/resolvers.py", line 731, in urlconf_module
    return import_module(self.urlconf_name)
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/me/devel/source/django-allauth/examples/regular-django/example/urls.py", line 9, in <module>
    admin.site.login = secure_admin_login(admin.site.login)
  File "/home/me/devel/source/django-allauth/examples/regular-django/venv/lib/python3.10/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/me/devel/source/django-allauth/examples/regular-django/venv/lib/python3.10/site-packages/django/utils/decorators.py", line 48, in _wrapper
    return bound_method(*args, **kwargs)
  File "/home/me/devel/source/django-allauth/examples/regular-django/venv/lib/python3.10/site-packages/allauth/decorators.py", line 10, in wrap
    resp = ratelimit.consume_or_429(request, action=action, **rl_kwargs)
  File "/home/me/devel/source/django-allauth/examples/regular-django/venv/lib/python3.10/site-packages/allauth/core/ratelimit.py", line 129, in consume_or_429
    if not consume(request, *args, **kwargs):
  File "/home/me/devel/source/django-allauth/examples/regular-django/venv/lib/python3.10/site-packages/allauth/core/ratelimit.py", line 95, in consume
    if not request or request.method == "GET":
AttributeError: 'function' object has no attribute 'method'
```

**Comment by rcj4747** [comment]
Sorry, I'm missing something.  I recreated this in a dirty environment (a 0.63.0 tree but 0.63.1 installed in the virtualenv).  I'll close this and debug my environment.

**Comment by pennersr** [comment]
The examples are using an outdated version in their requirements, fixed here: 1c8b41f6

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/3831
